5 create dog class

sunilsamra opened the pull request:
#5 
Finished the Dog Class ticket - created a Dog package and module. Wrote unit-tests to test Dog functionality

Commit:
#5
Created dog package structure and Dog class

Commit:
Merge remote-tracking branch 'origin/dev' into 5-create-dog-class

Commit:
#5
Completed Dog class with inheritance - added Dog unit tests

Commit:
Merge remote-tracking branch 'origin/dev' into 5-create-dog-class

ChrisMcCarthyDev commented on dog/dog.py:
```diff
@@ -0,0 +1,42 @@
+        super().__init__(species)
+        self.breed: Final[str] = breed
+
+    def __eq__(self, other):
```
We would need to override the `Animal.__hash__` function and make use of it here in the `Dog.__eq__` function.

sunilsamra replied:
I have overwritten the Animal.__hash__function to make use of it in the __eq__ function.

ChrisMcCarthyDev commented on tests/unit_tests/test_dog.py:
```diff
@@ -0,0 +1,23 @@
+    assert dog_1 == dog_2
+    assert dog_1 == dog_1
+
+
```
Once the `Dog.__hash__` function has been added and made use of in `Dog.__eq__`, you should add a test that tests comparing an instance of `Dog` against something that isn't an instance of `Dog` returns false.

sunilsamra replied:
I have added a new unit test to test a Dog instance is not the same as an Animal instance.

Commit:
#5
Updated Dog class with override __hash__ function and new unit tests against Dog and Animal classes being different

Commit:
#5
Updated Dog class with new comment

ChrisMcCarthyDev commented on dog/dog.py:
```diff
@@ -0,0 +1,53 @@
+        :return boolean: (True or False)
+        """
+        if isinstance(other, self.__class__):
+            return hash(self) == hash(other)
```
After line 37 we need to `return False`. This is just a way of eq always returning either True or False. We can have this default `return False` if we have determined that other is not an instance of `Dog`.

Commit:
#5
Updated Dog class with removal of __eq__ inherited method and shortened hash function
